components/ai_chat/core/browser/engine/engine_consumer_oai_unittest.cc

```diff
@@ -0,0 +1,190 @@
+
+    engine_->SetAPIForTesting(std::make_unique<MockOAIAPIClient>());
+  }
+
+  MockOAIAPIClient* GetClient() {
+    return static_cast<MockOAIAPIClient*>(engine_->GetAPIForTesting());
+  }
+
+  void TearDown() override {}
+
+ protected:
+  base::test::TaskEnvironment task_environment_;
+  mojom::ModelPtr model_;
+  std::unique_ptr<EngineConsumerOAIRemote> engine_;
+};
+
+TEST_F(EngineConsumerOAIUnitTest, GenerateQuestionSuggestions) {
+  std::string page_content = "This is a test page content";
+
+  auto* client = GetClient();
+  base::RunLoop run_loop;
+
+  auto invoke_completion_callback = [](const std::string& result_string) {
+    return [result_string](
+               const mojom::CustomModelOptions&, base::Value::List,
+               EngineConsumer::GenerationDataCallback,
+               EngineConsumer::GenerationCompletedCallback completed_callback) {
+      std::move(completed_callback)
+          .Run(EngineConsumer::GenerationResult(result_string));
+    };
+  };
+
+  EXPECT_CALL(*client, PerformRequest(_, _, _, _))
+      .WillOnce(invoke_completion_callback("Returning non question format"))
+      .WillOnce(invoke_completion_callback(
+          "<question>Question 1</question><question>Question 2</question>"))
+      .WillOnce(invoke_completion_callback(
+          "<question>Question 1</question>\n\n<question>Question 2</question>"))
+      .WillOnce(invoke_completion_callback(
+          "< question>>Question 1<</question><question>Question 2</question "
+          ">"));
+
+  engine_->GenerateQuestionSuggestions(
+      false, page_content,
+      base::BindLambdaForTesting(
+          [&](EngineConsumer::SuggestedQuestionResult result) {
+            EXPECT_TRUE(result.has_value());
+            EXPECT_EQ(result.value().size(), 1ull);
+          }));
+
+  engine_->GenerateQuestionSuggestions(
+      false, page_content,
+      base::BindLambdaForTesting(
+          [&](EngineConsumer::SuggestedQuestionResult result) {
+            EXPECT_STREQ(result.value()[0].c_str(), "Question 1");
+            EXPECT_STREQ(result.value()[1].c_str(), "Question 2");
+          }));
+
+  engine_->GenerateQuestionSuggestions(
+      false, page_content,
+      base::BindLambdaForTesting(
+          [&](EngineConsumer::SuggestedQuestionResult result) {
+            EXPECT_STREQ(result.value()[0].c_str(), "Question 1");
+            EXPECT_STREQ(result.value()[1].c_str(), "Question 2");
+          }));
+
+  engine_->GenerateQuestionSuggestions(
+      false, page_content,
+      base::BindLambdaForTesting(
+          [&](EngineConsumer::SuggestedQuestionResult result) {
+            EXPECT_STREQ(result.value()[0].c_str(), "Question 1");
+            EXPECT_STREQ(result.value()[1].c_str(), "Question 2");
+            run_loop.Quit();
+          }));
+
+  run_loop.Run();
+  testing::Mock::VerifyAndClearExpectations(client);
+}
+
+TEST_F(EngineConsumerOAIUnitTest, TestGenerateAssistantResponse) {
```

Comment:
The purpose of `GenerateAssistantResponse` is to take conversation entries and generate data that the OAI API is expecting. This test only seems to have a single expectation - that the first message will be a "system" role. Surely it needs to verify things like 1) it's included all the history items 2) that each history item looks as expected. See `EngineConsumerConversationAPIUnitTest` and all the `GenerateEvents_*` variations.

Reply:
ok, i am now testing individual history item and it's content.